Communication/Packets/Incoming/Groups/PostGroupContent.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Plus.Communication.Packets.Outgoing.Groups;
+using Plus.HabboHotel.GameClients;
+using Plus.HabboHotel.Groups.Forums;
+using System.Threading.Tasks;
+
+namespace Plus.Communication.Packets.Incoming.Groups;
+
+internal class PostGroupContentEvent : IPacketEvent
+{
+    private readonly IGroupForumManager _groupForumManager;
+    public PostGroupContentEvent(IGroupForumManager groupForumManager) => _groupForumManager = groupForumManager;
+    public async Task Parse(GameClient Session, ClientPacket Packet)
+    {
+        int ForumId = Packet.PopInt();
+        int ThreadId = Packet.PopInt();
+        string Caption = Packet.PopString();
+        string Message = Packet.PopString();
+
+        GroupForum Forum = _groupForumManager.GetForum(ForumId);
+        if (Forum == null)
+        {
+            return;
+        }
+
+        bool IsNewThread = ThreadId == 0;
+        
+        if (IsNewThread)
+        {
+
+            if ((Forum.Settings.GetReasonForNot(Session, Forum.Settings.WhoCanInitDiscussions)) != "")
+            {
+                Session.SendWhisper("Oops! Something went wrong. You do not have permission to start discussions here.");
+            }
+
+            GroupForumThread? Thread = await Forum.CreateThread(Session.GetHabbo().Id, Caption);
+            GroupForumThreadPost? Post = await Thread.CreatePost(Session.GetHabbo().Id, Message);
```

Comment:
`thread` is nullable but you dont check for that on line 36.

Are you not getting warnings / errors in your editor?